ccmlib/scylla_node.py

```diff
@@ -333,8 +339,19 @@ def start(self, join_ring=True, no_wait=False, verbose=False,
 
         if '--developer-mode' not in args:
             args += ['--developer-mode', 'true']
-        if '--smp' not in args:
-            args += ['--smp', '1']
+        if self._smp_set_during_test:
+            # If node.set_smp() is called during the test, ignore the --smp
+            # passed from the cmdline.
+            if '--smp' not in args:
+                args += ['--smp', str(self._smp)]
+            else:
+                args[args.index('--smp') + 1] = str(self._smp)
+        else:
+            # If node.set_smp() is not called during the test, use the --smp
+            # passed from the cmdline. If user does not pass --smp, default to
+            # --smp 1.
+            if '--smp' not in args:
+                args += ['--smp', '1']
```

Comment:
would this be simpler

 if '--smp' not in args:
   args += ['--smp', '1']
if self._smp_set_during_test:
    args[args.index('--smp') + 1] = str(self._smp)